src/Microsoft.OData.Client/XmlConstants.cs

```diff
@@ -982,6 +982,9 @@ internal static class XmlConstants
         /// <summary>edm string primitive type name</summary>
         internal const string EdmDateTimeOffsetTypeName = "Edm.DateTimeOffset";
 
+        /// <summary>edm string primitive type name</summary>
+        internal const string EdmDateTimeTypeName = "Edm.DateTime";

```

Comment:
OData V4 removed Edm.DateTime from the spec. So this is actually not a built-in primitive type. And this type will never  show in CSDL file. So I think it is better to change both the variable name and the constant string. Please also update the comment.